sources/osg/Uniform.js

```diff
@@ -26,395 +27,217 @@ Uniform.isUniform = function ( obj ) {
 
 /** @lends Uniform.prototype */
 Uniform.prototype = {
+
     getName: function () {
-        return this.name;
+        return this._name;
+    },
+
+    dirty: function () {
+        Notify.log( 'deprecated no need anymore' );
     },
+
     getType: function () {
-        return this.type;
+        return this._type;
     },
 
-    get: function () { // call dirty if you update this array outside
-        return this.data;
+    get: function () {
+        Notify.log( 'deprecated use getArray instead' );
+        return this._data;
     },
+
     set: function ( array ) {
-        this.data = array;
-        this.dirty();
-    },
-    dirty: function () {
-        this._dirty = true;
+        Notify.log( 'deprecated use setArray instead' );
+        this._data = array;
     },
+
     apply: function ( gl, location ) {
-        if ( this._dirty ) {
-            this.update.call( this.glData, this.data );
-            this._dirty = false;
-        }
-        this.glCall( gl, location, this.glData );
+        if ( this._isMatrix )
+            gl[ this._glCall ]( location, this._transpose, this._data );
+        else
+            gl[ this._glCall ]( location, this._data );
     },
-    applyMatrix: function ( gl, location ) {
-        if ( this._dirty ) {
-            this.update.call( this.glData, this.data );
-            this._dirty = false;
-        }
-        this.glCall( gl, location, this.transpose, this.glData );
+
+    setArray: function ( array ) {
```

Comment:
Idea to convey would be along the lines of setArrayDirect or setDirect, etc.
(as it does makes clearer the opposition between the others way that does need copy at each value change)